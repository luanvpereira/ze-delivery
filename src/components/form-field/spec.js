import React from 'react';
import { mount } from 'enzyme';

import FormField from './';

describe('<FormField/>', () => {
	it('should render a <FormField type="text">', () => {
		expect(
			mount(<FormField label="Nome" isLoading={true} />)
		).toMatchSnapshot();
	});

	it('should show search icon when pass isLoading attribute as true', () => {
		const component = mount(<FormField label="Nome" type="text" search />);

		expect(component.find('.formFieldSearch')).toHaveLength(1);
	});

	it('should render a <FormField type="select">', () => {
		expect(
			mount(
				<FormField type="select" label="opções">
					<option>Option 1</option>
					<option>Option 2</option>
				</FormField>
			)
		).toMatchSnapshot();
	});
});
