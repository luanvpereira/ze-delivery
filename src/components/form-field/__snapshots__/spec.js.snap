// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormField/> should render a <FormField type="select"> 1`] = `
<FormField
  label="opções"
  type="select"
>
  <div>
    <p
      className="formFieldLabel"
    >
      opções
    </p>
    <div
      className="formField"
    >
      <select
        className="formFieldSelect"
      >
        <option>
          Option 1
        </option>
        <option>
          Option 2
        </option>
      </select>
    </div>
  </div>
</FormField>
`;

exports[`<FormField/> should render a <FormField type="text"> 1`] = `
<FormField
  isLoading={true}
  label="Nome"
>
  <div
    className="formFieldIsLoading"
  >
    <p
      className="formFieldLabel"
    >
      Nome
    </p>
    <div
      className="formField"
    >
      <input
        className="formFieldInput"
      />
    </div>
  </div>
</FormField>
`;
